client/landing/gutenboarding/stores/domain-suggestions/selectors.ts

```diff
@@ -1,15 +1,63 @@
+/**
+ * External dependencies
+ */
+import { select } from '@wordpress/data';
+
 /**
  * Internal dependencies
  */
+import { STORE_KEY } from '.';
 import { DomainSuggestionQuery } from './types';
 import { State } from './reducer';
-import { getSerializedDomainsSuggestionsQuery } from 'state/domains/suggestions/utils';
+import { stringifyDomainQueryObject } from './utils';
 
 export const getState = ( state: State ) => state;
-export const getDomainSuggestions = ( state: State, queryObject: DomainSuggestionQuery ) => {
-	const serializedQuery = getSerializedDomainsSuggestionsQuery( queryObject );
-	if ( serializedQuery ) {
-		return state.domainSuggestions[ serializedQuery ];
-	}
-	return null;
+
+type DomainSuggestionSelectorOptions = Partial< Exclude< DomainSuggestionQuery, 'query' > >;
+export const getDomainSuggestions = (
+	state: State,
+	search: string,
+	options: DomainSuggestionSelectorOptions = {}
+) => {
+	const normalizedQuery = normalizeDomainSuggestionQuery( search, options );
+
+	// We need to go through the `select` store to get the resolver action
```

Comment:
Also means we shouldn't add a resolver for `getDomainSuggestions`, right? Can we somehow enforce that people don't (accidentally) do that?

Reply:
Other than documenting it or creating one that doesn't do anything, I can't think of a way to enforce it.